src/Microsoft.DotNet.Build.Tasks/FilterRuntimesFromSupports.cs

```diff
@@ -57,7 +59,7 @@ public static JObject GenerateCustomSupportsClause(List<TfmRidPair> applicableTf
             var tfmRidsForSupports = new JObject();
             foreach (var tfmRidPair in applicableTfmRidPairs)
             {
-                tfmRidsForSupports[tfmRidPair.framework] = new JArray(tfmRidPair.runtimeIdentifier);
+                tfmRidsForSupports[NuGetFramework.Parse(tfmRidPair.framework).GetShortFolderName()] = new JArray(tfmRidPair.runtimeIdentifier);
```

Comment:
I don't think you'll hit it since you are filtering to a single RID, but technically this could be overwriting entries in tfmRidsForSupports if you have multiple tuples with the same framework and different RIDs.

Also, I don't think this change is technically needed.  I think NuGet will handle the long form here.